### ISSUE TITLE
Editor Checkout - Focus management

### ISSUE THREAD

**Issue by ciampo** [Automattic/wp-calypso#46486]
#### Steps to reproduce

1. Open the editor for a page or a post of a free site.
2. Open up dev console, change the frame to `post.php`, and run the following code.
```js
let cartData = { products: [{
    product_id: 1009,
    product_slug: 'personal-bundle',
}] };

window.wp.hooks.doAction('a8c.wpcom-block-editor.openCheckoutModal', cartData);
```
3. Checkout modal should appear.
4. Notice how the focus is not moved inside the checkout when opening the checkout
5. Press tab and highlight all of the focusable elements in the checkout
6. Notice how, after tab is pressed on the last focusable element, the focus moves outside of the checkout modal again

#### What I expected

1. When opening the checkout modal, the focus should be moved to the modal itself
2. The focus should be trapped in the modal — ie:
  - pressing tab on the last focusable element should move the focus to the first focusable element
  - pressing shift+tab on the first focusable element should move the focus on the last focusable element

#### Browser / OS version

All browsers / OSs
